patches/api/0468-Add-listing-API-for-PlayerProfile.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Flo0 <flo.roma@web.de>
+Date: Sat, 6 Apr 2024 23:15:08 +0200
+Subject: [PATCH] Add listing API for PlayerProfile
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index c6cb4f17469a8f2e60dd3e28d41402851ce5fb21..8d35bb590facb83487729fe5cca9e8450bbac9d2 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -2054,6 +2054,69 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     boolean listPlayer(@NotNull Player other);
+     // Paper end
+ 
++    // Paper start - Add Listing API for PlayerProfile
++    /**
++     * Lists the {@code profile} in the tablist.
++     * Burden of tracking the profile is on the caller.
++     * Using profiles of online players is an invalid operation.
++     * Use {@link #listPlayer(Player)} instead.
++     *
++     * @param profile The {@link com.destroystokyo.paper.profile.PlayerProfile} to list.
++     * @param latency The latency of the profile in milliseconds.
++     * @param gameMode The game mode of the profile.
++     * @param displayName The display name of the profile.
++     */
++    void listProfile(@NotNull com.destroystokyo.paper.profile.PlayerProfile profile, int latency, @NotNull GameMode gameMode, @Nullable net.kyori.adventure.text.Component displayName);
++
++    /**
++     * Lists the {@code profile} in the tablist.
++     * Burden of tracking the profile is on the caller.
++     * Using profiles of online players is an invalid operation.
++     * Use {@link #listPlayer(Player)} instead.
++     *
++     * @param profile The {@link com.destroystokyo.paper.profile.PlayerProfile} to list.
++     */
++    default void listProfile(@NotNull com.destroystokyo.paper.profile.PlayerProfile profile) {
++        listProfile(profile, 0, GameMode.SURVIVAL, null);
++    }
```

Comment:
I'm not sure having this method makes much sense here. These defaults are a little arbitrary. Like 0 ping? why survival? what if its a creative server? I'm not sure we need anything except the full method with all the options.

Reply:
Those default values where taken from the internal EntryBuilder of the ClientboundPlayerInfoUpdatePacket.
Its honestly just a quol method. Would be nice to have a second opinion of this ^^